sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1086 @@
+#define IEEE802154_RADIO_H
+
+#include <stdbool.h>
+#include "iolist.h"
+#include "sys/uio.h"
+#include "byteorder.h"
+#include "net/eui64.h"
+
+/**
+ * @brief Forward declaration of the radio ops structure.
+ */
+typedef struct ieee802154_radio_ops ieee802154_radio_ops_t;
+
+typedef enum {
+    /**
+     * @brief the transceiver received a valid ACK with the frame pending bit
+     */
+    TX_FLAGS_FRAME_PENDING=(1<<0),
+    /**
+     * @brief the transceiver ran out of retransmission retries
+     */
+    TX_FLAGS_NO_ACK=(1<<1),
+    /**
+     * @brief the CSMA-CA algorithm or Auto CCA failed to measure a clear channel
+     */
+    TX_FLAGS_MEDIUM_BUSY=(1<<2),
+} ieee802154_tx_status_t;
+
+/**
+ * @brief IEEE802.15.4 Radio capabilities
+ * 
+ * These flags represent the hardware capabilities of a given device.
+ */
+typedef enum {
+    /**
+     * @brief the device supports hardware address filter
+     */
+    IEEE802154_CAP_HW_ADDR_FILTER,
+    /**
+     * @brief the device support frame retransmissions with CSMA-CA
+     */
+    IEEE802154_CAP_FRAME_RETRIES,
+    /**
+     * @brief the device support ACK timeout interrupt
+     */
+    IEEE802154_CAP_IRQ_ACK_TIMEOUT,
+    /**
+     * @brief the transceiver supports transmission with automatic CCA
+     */
+    IEEE802154_CAP_AUTO_CCA,
+    /**
+     * @brief the device supports Auto ACK
+     */
+    IEEE802154_CAP_AUTO_ACK,
+    /**
+     * @brief the device supports the IEEE802.15.4 2.4 GHz band
+     */
+    IEEE802154_CAP_24_GHZ,
+    /**
+     * @brief the device support the IEEE802.15.4 Sub GHz band
+     */
+    IEEE802154_CAP_SUB_GHZ,
+    /**
+     * @brief the device reports when the transmission is done
+     */
+    IEEE802154_CAP_IRQ_TX_DONE,
+    /**
+     * @brief the device reports the start of a frame (SFD) when received.
+     */
+    IEEE802154_CAP_IRQ_RX_START,
+    /**
+     * @brief the device reports the end of the CCA procedure
+     */
+    IEEE802154_CAP_IRQ_CCA_DONE,
+    /**
+     * @brief the device supports going to RX after TX done.
+     */
+    IEEE802154_CAP_TXRX,
+    /**
+     * @brief the device provides the number of retransmissions
+     */
+    IEEE802154_CAP_FRAME_RETRIES_INFO,
+} ieee802154_rf_caps_t;
+
+/**
+ * @brief IEEE802.15.4 transceiver states
+ */
+typedef enum {
+    IEEE802154_TRX_STATE_TRX_OFF, /**< the transceiver state if off */
+    IEEE802154_TRX_STATE_RX_ON, /**< the transceiver is ready to receive packets */
+    IEEE802154_TRX_STATE_TX_ON, /**< the transceiver is ready to transmit a frame */
+} ieee802154_trx_state_t;
+
+/**
+ * @brief IEEE802.15.4 Radio HAL events
+ *
+ * To follow the IEEE802.15.4 convention, an event that responds to a Request
+ * is a confirmation (Confirm). Otherwise an Indication.
+ */
+typedef enum {
+    /**
+     * @brief the transceiver detected a valid SFD
+     */
+    IEEE802154_RADIO_INDICATION_RX_START,
+
+    /**
+     * @brief the transceiver received a frame and there's a frame in the
```

Comment:
Guess it is the same frame that has been received which lies in the buffer? The "and" confuses me a bit.

Reply:
yes, I think "lies in the framebuffer" is more accurate

Reply:
https://github.com/RIOT-OS/RIOT/pull/14371/commits/f6b8f96c21589f0534f1e3ad9f678579e9ea3cba